README.md

```diff
@@ -26,6 +26,55 @@ An admin tool that allows you to control a block layout across units.
  $CFG->blockmanagerclass = '\\tool_blocksmanager\\block_manager';
 ```
 
+## Usage ##
+
+### Block management ###
+This feature allows to lock a specific block in a specific region through a selected course category so no one (expect users with 
+'tool/blocksmanager:bypasslocking') can apply changes to a block configuration in t hat region.
+
+Actions that could be locked:
+- configuration
+- deleting
+- hiding 
+- moving
+
+To create set of rules to lock blocks, navigate to Site administration > Plugins > Admin tools > Blocks Manager > Blocks management    
+
+### Region management ###
+This feature gives an ability to lock a whole region through a selected course category. That means all blocks in the 
+matching region will be locked from performing configured actions (expect for users with 'tool/blocksmanager:bypasslocking'). 
+
+Actions that could be locked:
+- configuration
+- deleting
+- hiding
+- moving
+
+To create set of rules to lock regions, navigate to Site administration > Plugins > Admin tools > Blocks Manager > Region management
+
+### Set up blocks ### 
+This features helps to create blocks in bulk through selected course categories (for example add a calendar block to every course 
+for a selected category).
+
+To do  that, navigate to Site administration > Plugins > Admin tools > Blocks Manager > Set up blocks
+
+The form let you to select: 
+
+- Region - what region to add a block to?
+- Block - what block to add? 
+- Categories - what categories to apply the block? 
+- Weight - what block weight will be on those pages? 
+- Config data - does block have any config? Could be applied here (but should be exact string of config that is stored in DB if you would add the block manually)? 
+- Visible? - is block visible when added? 
+- Add another instance (if exists)? - if the ame block already exists, should we add another instance of that block? 
+- Update instance (if exists)? - if the block already exists, should we update the block by new settings? 
+- Reposition instance (if exists)? - if the block already exists, should we reposition the block?
+- Show in subcontexts - should the block be displayed on course subpages (e.g. course page and activity pages)?
+- Page type pattern - what page should the block be displayed? E.g. course-view-*, mod-assign-view and etc.
+                                                                                                           
+Multiple blocks can be configured to be applied. Once all done and form is submitted. The set of blocks will be applied 
```

Comment:
I was confused about `Add new line` and `Apply set of blocks` button.
Also better to add what `Set of blocks` textarea is.

Reply:
Yeah. I know UI is confusing. I tried to improve the description, please check.